packages/lab/stories/date-picker/date-picker.stories.tsx

```diff
@@ -1,209 +1,1099 @@
+        }}
+      >
+        <DatePickerRangeInput
+          onDateValueChange={(newDateValue?: DateInputRangeValue) => {
+            const validationStatus =
+              isValidNumericDate(newDateValue?.startDate, "DD/MM/YYYY") &&
+              isValidNumericDate(newDateValue?.endDate, "DD/MM/YYYY")
+                ? undefined
+                : "error";
+            setValidationStatus(validationStatus);
+          }}
+          formatDate={formatDateStringEsES}
+          parse={parseDateStringEsES}
+          placeholder={"DD/MM/YYYY"}
+        />
+        <DatePickerOverlay>
+          <DatePickerRangePanel />
+        </DatePickerOverlay>
+      </DatePicker>
+      <FormHelperText>{helperText}</FormHelperText>
+    </FormField>
+  );
+};
+
+export const SingleBordered: StoryFn<DatePickerSingleProps> = (args) => {
   return (
     <DatePicker
       {...args}
-      selectedDate={selectedDate}
-      onSelectionChange={(_, date) => setSelectedDate(date as DateValue)}
-    />
+      selectionVariant="single"
+      onSelectedDateChange={(newSelectedDate) => {
+        console.log(`Selected date: ${formatDate(newSelectedDate)}`);
+        args?.onSelectedDateChange?.(newSelectedDate);
+      }}
+    >
+      <DatePickerSingleInput bordered />
+      <DatePickerOverlay>
+        <DatePickerSinglePanel
+          NavigationProps={{
+            MonthDropdownProps: { bordered: true },
+            YearDropdownProps: { bordered: true },
+          }}
+        />
+      </DatePickerOverlay>
+    </DatePicker>
   );
 };
 
-export const RangeControlled: StoryFn<
-  DatePickerProps<RangeSelectionValueType>
-> = (args) => {
-  const [selectedDate, setSelectedDate] = useState<
-    RangeSelectionValueType | undefined
-  >(undefined);
+export const RangeBordered: StoryFn<DatePickerRangeProps> = (args) => {
   return (
     <DatePicker
       {...args}
       selectionVariant="range"
-      selectedDate={selectedDate}
-      onSelectionChange={(_, date) => {
-        setSelectedDate(date as RangeSelectionValueType);
+      onSelectedDateChange={(newSelectedDate) => {
+        console.log(`Selected date range: ${formatDateRange(newSelectedDate)}`);
+        args?.onSelectedDateChange?.(newSelectedDate);
       }}
-    />
+    >
+      <DatePickerRangeInput bordered />
+      <DatePickerOverlay>
+        <DatePickerRangePanel
+          StartNavigationProps={{
+            MonthDropdownProps: { bordered: true },
+            YearDropdownProps: { bordered: true },
+          }}
+          EndNavigationProps={{
+            MonthDropdownProps: { bordered: true },
+            YearDropdownProps: { bordered: true },
+          }}
+        />
+      </DatePickerOverlay>
+    </DatePicker>
   );
 };
 
-export const ControlledOpenOnEnter: StoryFn<
-  DatePickerProps<SingleSelectionValueType | RangeSelectionValueType>
-> = (args) => {
-  // state
-  const [open, setOpen] = useState(false);
+const DatePickerTimeInput: React.FC = () => {
```

Comment:
Not sure what's intended example here. If I change time input from using mouse via popup, i can see selection change fired, but not seeing time input change (still `--:--`) nor seeing date range pick change (are we expecting time shown on it?)